addons/l10n_sa_edi/models/account_edi_format.py

```diff
@@ -476,3 +479,38 @@ def _get_move_applicability(self, move):
             'post': self._l10n_sa_post_zatca_edi,
             'edi_content': self._l10n_sa_get_invoice_content_edi,
         }
+
+    def _prepare_invoice_report(self, pdf_writer, edi_document):
+        """
+        Prepare invoice report to be printed.
+        :param pdf_writer: The pdf writer with the invoice pdf content loaded.
+        :param edi_document: The edi document to be added to the pdf file.
+        """
+        super()._prepare_invoice_report(pdf_writer, edi_document)
+        if self.code != 'sa_zatca' or edi_document.move_id.country_code != 'SA':
+            return
+
+        attachment = edi_document.attachment_id
+        if not attachment or not attachment.datas:
+            _logger.warning(f"No attachment found for invoice {edi_document.move_id.name}")
+            return
+
+        # Convert attachment data from base64 to raw bytes.
+        xml_content = b64decode(attachment.datas)
```

Comment:
Why don't we use `.raw` here?
```suggestion
        xml_content = attachment.raw
```

https://github.com/odoo/odoo/blob/a758a83dcd8795cf4c4f154a56cf4a86dbeb0567/odoo/addons/base/models/ir_attachment.py#L414-L415